skfda/exploratory/visualization/representation.py

```diff
@@ -1,32 +1,80 @@
+"""Representation Module.
+
+This module contains the functionality related
+with plotting and scattering our different datasets.
+It allows multiple modes and colors, which could
+be set manually or automatically depending on values
+like depth measures.
+"""
+
+from typing import (
+    Any,
+    List,
+    Mapping,
+    Optional,
+    Sequence,
+    Tuple,
+    TypeVar,
+    Union,
+)
 
 import matplotlib.cm
 import matplotlib.patches
 import numpy as np
+from matplotlib.axes import Axes
+from matplotlib.figure import Figure
+from typing_extensions import Protocol
 
+from ... import FDataGrid
 from ..._utils import _to_domain_range, constants
+from ...representation._functional_data import FData
+from ...representation._typing import DomainRangeLike
 from ._utils import (
+    ColorLike,
     _get_figure_and_axes,
     _set_figure_layout_for_fdata,
     _set_labels,
 )
 
+K = TypeVar('K', contravariant=True)
+V = TypeVar('V', covariant=True)
+T = TypeVar('T', FDataGrid, np.ndarray)
+
+
+class Indexable(Protocol[K, V]):
+    """Class Indexable used to type _get_color_info."""
 
-def _get_label_colors(n_labels, group_colors=None):
-    """Get the colors of each label"""
+    def __getitem__(self, __key: K) -> V:
+        pass
 
+
+def _get_label_colors(
+    n_labels: int,
+    group_colors: Union[Sequence[ColorLike], Mapping[K, ColorLike], None],
+) -> np.ndarray:
+    """Get the colors of each label."""
     if group_colors is not None:
```

Comment:
Maybe it is also better to reverse the order of the `if-else` in order to use positive conditions.